config/routes.rb

```diff
@@ -14,17 +14,17 @@
   post '/api/irs_attempts_api/security_events' => 'api/irs_attempts_api#create'
 
   # SAML secret rotation paths
-  SamlEndpoint.suffixes.each do |suffix|
-    get "/api/saml/metadata#{suffix}" => 'saml_idp#metadata', format: false
-    match "/api/saml/logout#{suffix}" => 'saml_idp#logout', via: %i[get post delete]
-    match "/api/saml/remotelogout#{suffix}" => 'saml_idp#remotelogout', via: %i[get post]
+  constraints(path_year: SamlEndpoint.suffixes) do
```

Comment:
This is a pretty cool feature I wasn't aware of, but I'm at a loss for how it even works. I don't see any documentation for it, and the closest I got to thinking I understood it was thinking it was implicitly calling `Array#matches?`, but `matches?` doesn't appear to be a method on an array.

Reply:
In the Rails Routing guide, I pieced this together from [segment constraints](https://guides.rubyonrails.org/routing.html#segment-constraints) and [the constraints block form](https://guides.rubyonrails.org/routing.html#specifying-constraints) in

So the block form is just a way to avoid repeating the segment constraints for each line

Reply:
and in terms of implementation, I bet that it's using our favorite `===` which for strings and regexes is matches, but for arrays is "contains"